src/kafunk/Consumer.fs

```diff
@@ -800,9 +800,8 @@ module Consumer =
       c.conn.Config.connId cfg.groupId topic (Printers.partitions assignment)
       
     if assignment.Length = 0 then
-      Log.error "no_partitions_assigned|conn_id=%s group_id=%s member_id=%s topic=%s" 
+      Log.warn "no_partitions_assigned|conn_id=%s group_id=%s member_id=%s topic=%s" 
         c.conn.Config.connId cfg.groupId state.state.memberId topic
-      return failwithf "no partitions assigned!"
```

Comment:
I think you should do `return [||]` here rather than just letting the continuation figure that out. Also, have you tested this with Consumer.fsx?

Reply:
Yup tested with Consumer.fsx, and it runs. I tried using the `return [||]` as suggested in #186 but the compiler fails because the log statement is in an else-less `if`, requiring a unit to be returned from the expression.

Reply:
Try this `return [||] else`.

Reply:
Done, also moved the else branch to separate private function for readability.